src/content/ui-api/management-api/wc-key-pairs/index.md

```diff
@@ -0,0 +1,264 @@
+---
+linkTitle: Key pairs for workload clusters
+title: Creating workload cluster key pairs via the Management API
+description: We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a Giant Swarms specific way to create such key pairs via the Management API.
+weight: 50
+menu:
+  main:
+    identifier: uiapi-managementapi-keypairs
+    parent: uiapi-managementapi
+user_questions:
+  - How to create workload cluster key pairs via the Management API?
+last_review_date: 2021-07-26
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-biscuit
+---
+
+# Creating workload cluster key pairs via the Management API
+
+We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a method specific to Giant Swarm to create such key pairs via the Management API.
+
+**Note:** The method described here makes use of the [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) <abbr title="custom resource definition">CRD</abbr></a> and [cert-operator](https://github.com/giantswarm/cert-operator), which are part of every Giant Swarm management cluster currently. As Giant Swarm is shifting towards using Cluster API controllers, this method is not guaranteed to work once this transition is done. Please consider it a temporary solution.
+
+**Note:** If you want to use the deprecated Giant Swarm Rest API for creating key pairs, please check [`gsctl create kubeconfig`]({{< relref "/ui-api/gsctl/create-kubeconfig.md" >}}) and [`gsctl create keypair`]({{< relref "/ui-api/gsctl/create-keypair.md" >}}).
+
+## Prerequisites
+
+You need:
+
+- Your installation's Management API endpoint (similar to `https://g8s.codename.eu-west-1.aws.gigantic.io`)or base domain (similar to `codename.eu-west-1.aws.gigantic.io`).
+- Management API access as an admin.
+- Unique ID of the workload cluster to grant access to.
+- Name of the organization "owning" that workload cluster.
+
+These tools are also required:
+
+- `kubectl`
+- [kubectl gs]({{< relref "/ui-api/kubectl-gs/" >}}), the `kubectl` plug-in for the Giant Swarm Management API.
+
+If you want to use our [example script](#script), you also need:
+
+- [jq](https://stedolan.github.io/jq/)
+- The shell commands/utilities `base64`, `cut`, `date`, and `shasum`
+
+## Ensure Management API access
+
+Using our `kubectl` plug-in, you can make sure to be authenticated against the Management API:
+
+```nohighlight
+kubectl gs login <URL>
+```
+
+Here, `<URL>` can be either
+
+- the Management API endpoint URL or host name
```

Comment:
what does 'URL or host name' mean here?

Reply:
It means you can use either `https://blah` or `blah`